catalog/spatial/csw/spatial-csw-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/GetRecordsRequest.java

```diff
@@ -379,11 +379,17 @@ public GetRecordsType get202RecordsType() throws CswException {
                 queryConstraint.setCqlText(getConstraint());
             } else if (getConstraintLanguage().equalsIgnoreCase(CswConstants.CONSTRAINT_LANGUAGE_FILTER)) {
                 try {
+                    XMLInputFactory xmlInputFactory = XMLInputFactory.newFactory();
+                    xmlInputFactory.setProperty(XMLInputFactory.IS_SUPPORTING_EXTERNAL_ENTITIES,
+                            false);
+                    xmlInputFactory.setProperty(XMLInputFactory.SUPPORT_DTD, false);
+                    XMLStreamReader xmlStreamReader =
+                            xmlInputFactory.createXMLStreamReader(new StringReader(constraint));
+
                     Unmarshaller unmarshaller = JAX_BCONTEXT.createUnmarshaller();
```

Comment:
It looks like this should just be using our platform parser api instead of creating its own unmarshaller.